draft-rescorla-tls-esni.md

```diff
@@ -365,6 +369,66 @@ has cleaned out most such proxies.
 
 ## Comparison Against Criteria
 
+{{?I-D.ietf-tls-sni-encryption}} lists several requirements for SNI encryption. In this 
+section, we re-iterate these requirements and assess the ESNI design against them.
+
+1. Mitigate against replay attacks
+
+Since K_sni is derived from a (EC)DH operation between the client's ephemeral
+and server's semi-static ESNI key. This binds the ESNI encryption to the 
+Client Hello. It is not possible for an attacker to "cut and paste" the ESNI
+value in a different Client Hello, with a different ephemeral key share, as
+the terminating server will fail to decrypt and verify the ESNI value.
+
+2. Avoid widely-deployed shared secrets
+
+This design depends upon DNS as a vehicle for semi-static public key distribution.
+Server operators may partition their private keys however they see fit provided
+each server behind an IP address has the corresponding private key to decrypt 
+a key. Thus, when one ESNI key is provided, sharing is optimally bound by the number 
+of hosts that share an IP address. Server operators may further limit sharing
+by including multiple keys, with distinct labels, in an ESNIKeys structure.
+
+3. Prevent SNI-based DoS attacks
+
+This design requires servers to decrypt ClientHello messages with EncryptedSNI
+extensions carrying valid labels. Thus, it is possible for an attacker to force
+decryption operations on the server. Servers SHOULD NOT be configured to handle
+multiple keys, as this then require servers to perform trial decryption before
```

Comment:
Why is this a problem? You have a label?

Reply:
Oh, yes, oops. I was forgot we said above just to fail open if there's a label mismatch.